pkgs/build-support/fetchurl/default.nix

```diff
@@ -200,17 +200,17 @@ lib.extendMkDerivation {
         else if sha512 != "" then
           {
             outputHashAlgo = "sha512";
-            outputHash = sha512;
+            outputHash = finalAttrs.sha512;
           }
         else if sha256 != "" then
           {
             outputHashAlgo = "sha256";
-            outputHash = sha256;
+            outputHash = finalAttrs.sha256;
```

Comment:
That would make it impossible to override with `hash` if someone uses `sha256` during construction.